Source/NetTiers.cst

```diff
@@ -1505,10 +1505,12 @@
 
                     // Now that the sql has been executed, refresh the view meta data in case in changed.
                     SourceDatabase.Refresh();
+                    
+                    ViewSchemaCollection temp_templateSourceViews = templateSourceViews;
                     templateSourceViews = new ViewSchemaCollection();
-                    for (int i=0; i < SourceDatabase.Views.Count; i++)
+                    for (int i=0; i < temp_templateSourceViews.Count; i++)

```

Comment:
lets change all this code to be 

` foreach(var view in templateSourceViews.ToList()) {...}` that way it's a copy and cleaner. So all this is doing is refreshing the view logic? Are we sure this doesn't happen automatically by reference when we call SourceDatabase.Refresh()?

Reply:
Yes, I tried SourceDatabase.Refresh by itself.  That does seem to be necessary to update the SourceDatabase.Views, however there is a disconnect between the SourceDataBase views and the views the user has chosen.  The chosen views do not get updated.  My guess is that has something to do with how some of the automatic user parameter variables get created and mapped inside of Code Smith.

Reply:
what you have works, I just didn't understand why we were creating a new temp variable, I figured it was due to mutation of the original which could be fixed with a ToList().. That's really weird, I'd of thought the views would be updated by reference.

Reply:
I just submitted the cleaned up version.